test/units/parsing/yaml/test_dumper.py

```diff
@@ -59,3 +61,36 @@ def test(self):
         data_from_yaml = loader.get_single_data()
 
         self.assertEqual(plaintext, data_from_yaml.data)
+
+    def test_bytes(self):
+        b_text = u'tréma'.encode('utf-8')
+        unsafe_object = AnsibleUnsafeBytes(b_text)
+        yaml_out = self._dump_string(unsafe_object, dumper=self.dumper)
+
+        stream = self._build_stream(yaml_out)
+        loader = self._loader(stream)
+
+        data_from_yaml = loader.get_single_data()
+
+        result = b_text
+        if PY2:
+            # https://pyyaml.org/wiki/PyYAMLDocumentation#string-conversion-python-2-only
+            # pyyaml on Python 2 returns unicode and not bytes when given bytes with non-ASCII characters.
+            # Changes need to be made to the methods in Ansible to coerce the strings to behave consistently
+            # on Python 2 and Python 3.
+            # This is a workaround for the current behavior.
```

Comment:
Little more accurate version of this comment
```
            # https://pyyaml.org/wiki/PyYAMLDocumentation#string-conversion-python-2-only
            # pyyaml on Python 2 can return either unicode or bytes when given byte strings.
            # We normalize that to always return unicode on Python2 as that's right most of the
            # time.  However, this means byte strings can round trip through yaml on Python3 but
            # not on Python2.  To make this code work the same on Python2 and Python3 (we want
            # the Python3 behaviour) we need to change the methods in Ansible to:
            # (1) Let byte strings pass through on Python2 and
            # (2) Convert byte strings to text strings before being given to pyyaml
            # This is a workaround for the current behavior.
```

Reply:
Thanks!